### DISCUSSION TITLE
The optimization speed of flow field inversion is very slow

### DISCUSSION THREAD

**Discussion by 15175829310** [mdolab/dafoam#676] [General]
1、I am conducting flow field inversion, but the objective function is decreasing slowly and the magnitude of the decrease is also small. May I ask which parameters can be adjusted to accelerate the optimization progress.
2、During the calculation process, it was found that the calculation ended with only "TIME=1". Is this normal? If it is not normal, what parameters can be adjusted?
[logOpt.txt](https://github.com/user-attachments/files/16838452/logOpt.txt)
[opt_IPOPT.txt](https://github.com/user-attachments/files/16838453/opt_IPOPT.txt)
[runScript_v2.txt](https://github.com/user-attachments/files/16838454/runScript_v2.txt)

![ipopt](https://github.com/user-attachments/assets/315d7209-526b-4724-a420-bfc37e7c66a9)
![time](https://github.com/user-attachments/assets/ad833ec6-3f91-4624-afc6-da8bdccbe161)

**Comment by friedenhe** [comment]
Check my answer to your other question. The problem was caused by a bug in the residual calculation. Please use the latest version.

**Comment by NeuroNexusAI** [comment]
To speed up optimization, try adjusting the step size, convergence tolerance, or algorithm settings.

If the calculation ends with "TIME=1," check for max iterations, errors in the log files, and resource limits.

**Reply by ThinkingVonCN** [reply]
Excuse, I want know how to adjust step size, I have encounter the same question like this one. In my case, objective value decay very slow, it only decay 0.0001in one time. Please help me, thanks.
@friedenhe

**Comment by friedenhe** [comment]
@ThinkingVonCN , I believe @NeuroNexusAI is an AI bot, so you can probably ignore its answer. Regarding your Q, you probably need to check a few things such as whether the adjoint gradient is accurate, whether your flow/adjoint is converging. Other than changing step size.
